Neos.ContentRepository/Classes/Security/Authorization/Privilege/Node/NodePrivilegeContext.php

```diff
@@ -159,6 +176,42 @@ public function isInWorkspace($workspaceNames)
         return in_array($this->node->getWorkspace()->getName(), $workspaceNames);
     }
 
+    /**
+     * Matches if the currently-selected workspace is the target workspace of the current user.
+     *
+     * Example: userIsInTargetWorkspace(['preview-1234']) matches if the current user
+     * has selected workspace "preview-svxg3" as target workspace
+     *
+     * @param string|array $workspaceNames An array of workspace names, e.g. ["live", "preview-svxg3"]
+     * @return boolean true if the current user is in one of the given $workspaceNames, otherwise false
+     */
+    public function userIsInTargetWorkspace($workspaceNames): bool
+    {
+        if ($this->node === null) {
+            return true;
```

Comment:
That seems dangerous to me, but it's the same in the other matchers so it's probably fine to keep it here as well

Reply:
This might really be a case, where accurate tests are needed. I wondered about that one, too.